Added Discord notification for "@crash site" when map restarts

Jayefuu opened the pull request:
Added discord notification for the @crash site role when crash site is restarted using the /restart command. Pings users with @Crash Site role and prints the map time to add to competitiveness.

Jayefuu committed:
Added Discord notification for "@crash site" when map restarts

Added discord notification for the @crash site role when crash site is restarted using the /restart command. Pings users with @Crash Site role and prints the map time to add to competitiveness.

grilledham commented on map_gen/maps/crash_site/commands.lua:
```diff
@@ -42,6 +42,10 @@ callback =
         elseif state == 1 then
+            local game_seconds = math.floor(game.ticks_played / 60) -- should probably account for game speed changes, but we rarely (if ever?) change crash site performance scale. This is good enough?
+            local game_hours = math.floor(game_seconds / 3600)
+            local game_min_remainder = math.floor((game_seconds - (game_hours * 3600)) / 60)
```
Does this do what you want <https://github.com/Refactorio/RedMew/blob/develop/utils/core.lua#L136>?

grilledham commented on map_gen/maps/crash_site/commands.lua:
```diff
@@ -42,6 +42,10 @@ callback =
+            local game_seconds = math.floor(game.ticks_played / 60) -- should probably account for game speed changes, but we rarely (if ever?) change crash site performance scale. This is good enough?
+            local game_hours = math.floor(game_seconds / 3600)
+            local game_min_remainder = math.floor((game_seconds - (game_hours * 3600)) / 60)
+            Server.to_discord_raw('<@&762441731194748958> Crash Site has just restarted!! Previous map lasted: ' .. game_hours .. ' hours, ' .. game_min_remainder .. ' minutes')
```
Add markdown to make it **bold**?

```suggestion
            Server.to_discord_raw('<@&762441731194748958> **Crash Site has just restarted!! Previous map lasted: ' .. game_hours .. ' hours, ' .. game_min_remainder .. ' minutes**')
```

grilledham replied:
It might also be woth having a local variable for `local crashsite_role ='<@&762441731194748958>'` just to explain what that number is.

Jayefuu replied:
All good suggestions, thank you. I'll make the changes and test tonight.

Jayefuu replied:
I think I'll also try work out how to put the notification in #map-update not #s7 so that all the map pings stay in one place and people don't have to scroll through the server channel.

grilledham reviewed (approved):
Some suggestions, but this looks fine to me.

Jayefuu committed:
Update commands.lua

Added changed requested by grilledham:
- Used format_time function
- Added local var to show that text string was role ID.
- Added another commented var with the ID of a test role in case we want to test something and not ping @crash_site
- Added markdown to make printed string bold

grilledham commented on map_gen/maps/crash_site/commands.lua:
```diff
@@ -123,6 +128,7 @@ local function restart(args, player)
     end
     print('Abort restart with /abort')
 
+    --Server.to_discord_raw('<@&762441731194748958> Crash site is restarting!')
```
After our discssion about not being able to send the pings to the map update channel (yet), I assume you will uncomment this.

Jayefuu replied:
No, good catch. The first time I wrote this I had the ping go out when the restart command was called, but later moved it to the same place as the popup so that players didn't get a notification if /restart then /abort was used. I've now deleted that commented line since the to_discord_raw is now on line 49.

Jayefuu committed:
Update commands.lua
